make at Ubuntu 21.10 impish fails

```
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 21.10
Release:	21.10
Codename:	impish
```

----

So I'm running a fully uptodate Ubuntu 21.10 on AMD Ryzen 7

I completed the pre-reqs

I also installed fine   SysinternalsEBPF

my make i.e. last line fails thus

```

sudo make clean
m@lenny:~/code/SysmonForLinux/build$ sudo make
[  1%] Extracting sysmonmsg.mc.utf16
[  2%] Extracting sysmonmsgop.man.utf16
[  3%] Converting sysmonmsgop.man.utf16 to UTF8
[  4%] Extracting sysmonmsgop.c from sysmonmsgop.man
[  5%] Extracting sysmonevents.h.utf16
Warning skip check of User due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of User due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of User due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of User due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of CommandLine due to misalignment
Warning skip check of CurrentDirectory due to misalignment
Warning skip check of User due to misalignment
Warning skip check of LogonGuid due to misalignment
Warning skip check of LogonId due to misalignment
Warning skip check of TerminalSessionId due to misalignment
Warning skip check of IntegrityLevel due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of ParentProcessGuid due to misalignment
Warning skip check of ParentProcessId due to misalignment
Warning skip check of ParentImage due to misalignment
Warning skip check of ParentCommandLine due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Signed due to misalignment
Warning skip check of Signature due to misalignment
Warning skip check of SignatureStatus due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of CommandLine due to misalignment
Warning skip check of CurrentDirectory due to misalignment
Warning skip check of User due to misalignment
Warning skip check of LogonGuid due to misalignment
Warning skip check of LogonId due to misalignment
Warning skip check of TerminalSessionId due to misalignment
Warning skip check of IntegrityLevel due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of ParentProcessGuid due to misalignment
Warning skip check of ParentProcessId due to misalignment
Warning skip check of ParentImage due to misalignment
Warning skip check of ParentCommandLine due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Signed due to misalignment
Warning skip check of Signature due to misalignment
Warning skip check of SignatureStatus due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of FileVersion due to misalignment
Warning skip check of Description due to misalignment
Warning skip check of Product due to misalignment
Warning skip check of Company due to misalignment
Warning skip check of CommandLine due to misalignment
Warning skip check of CurrentDirectory due to misalignment
Warning skip check of User due to misalignment
Warning skip check of LogonGuid due to misalignment
Warning skip check of LogonId due to misalignment
Warning skip check of TerminalSessionId due to misalignment
Warning skip check of IntegrityLevel due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of ParentProcessGuid due to misalignment
Warning skip check of ParentProcessId due to misalignment
Warning skip check of ParentImage due to misalignment
Warning skip check of ParentCommandLine due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of CreationUtcTime due to misalignment
Warning skip check of PreviousCreationUtcTime due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of User due to misalignment
Warning skip check of Protocol due to misalignment
Warning skip check of Initiated due to misalignment
Warning skip check of SourceIsIpv6 due to misalignment
Warning skip check of SourceIp due to misalignment
Warning skip check of SourceHostname due to misalignment
Warning skip check of SourcePort due to misalignment
Warning skip check of SourcePortName due to misalignment
Warning skip check of DestinationIsIpv6 due to misalignment
Warning skip check of DestinationIp due to misalignment
Warning skip check of DestinationHostname due to misalignment
Warning skip check of DestinationPort due to misalignment
Warning skip check of DestinationPortName due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ImageLoaded due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Signed due to misalignment
Warning skip check of Signature due to misalignment
Warning skip check of SignatureStatus due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of ImageLoaded due to misalignment
Warning skip check of FileVersion due to misalignment
Warning skip check of Description due to misalignment
Warning skip check of Product due to misalignment
Warning skip check of Company due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Signed due to misalignment
Warning skip check of Signature due to misalignment
Warning skip check of SignatureStatus due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of SourceProcessGuid due to misalignment
Warning skip check of SourceProcessId due to misalignment
Warning skip check of SourceImage due to misalignment
Warning skip check of TargetProcessGuid due to misalignment
Warning skip check of TargetProcessId due to misalignment
Warning skip check of TargetImage due to misalignment
Warning skip check of NewThreadId due to misalignment
Warning skip check of StartAddress due to misalignment
Warning skip check of StartModule due to misalignment
Warning skip check of StartFunction due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of Device due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of SourceProcessGUID due to misalignment
Warning skip check of SourceProcessId due to misalignment
Warning skip check of SourceThreadId due to misalignment
Warning skip check of SourceImage due to misalignment
Warning skip check of TargetProcessGUID due to misalignment
Warning skip check of TargetProcessId due to misalignment
Warning skip check of TargetImage due to misalignment
Warning skip check of GrantedAccess due to misalignment
Warning skip check of CallTrace due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of CreationUtcTime due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetObject due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetObject due to misalignment
Warning skip check of Details due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetObject due to misalignment
Warning skip check of NewName due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of CreationUtcTime due to misalignment
Warning skip check of Hash due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of PipeName due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of Operation due to misalignment
Warning skip check of User due to misalignment
Warning skip check of EventNamespace due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Query due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of Operation due to misalignment
Warning skip check of User due to misalignment
Warning skip check of Name due to misalignment
Warning skip check of Type due to misalignment
Warning skip check of Destination due to misalignment
Warning skip check of EventType due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of Operation due to misalignment
Warning skip check of User due to misalignment
Warning skip check of Consumer due to misalignment
Warning skip check of Filter due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of TargetFilename due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Archived due to misalignment
Warning skip check of UtcTime due to misalignment
Warning skip check of ProcessGuid due to misalignment
Warning skip check of ProcessId due to misalignment
Warning skip check of Image due to misalignment
Warning skip check of Session due to misalignment
Warning skip check of ClientInfo due to misalignment
Warning skip check of Hashes due to misalignment
Warning skip check of Path due to misalignment
Warning skip check of Archived due to misalignment
[  6%] Converting sysmonevents.h.utf16 to UTF8
[  7%] Converting sysmonmsg.mc.utf16 to UTF8
[  8%] Extracting sysmonmsg.h from sysmonmsg.mc
[  9%] Extracting sysmonmsgop.h from sysmonmsgop.man
Scanning dependencies of target sysmonLogView
[ 10%] Building CXX object CMakeFiles/sysmonLogView.dir/sysmonLogView/sysmonLogView.cpp.o
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp: In function ‘void processCmdline(int, char**, std::unordered_set<unsigned int>*, std::unordered_map<unsigned int, std::unordered_set<std::__cxx11::basic_string<char> > >*, std::pair<long unsigned int, long unsigned int>*, std::pair<long unsigned int, long unsigned int>*, std::unordered_map<std::__cxx11::basic_string<char>, std::unordered_set<std::__cxx11::basic_string<char> > >*, bool*)’:
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:452:33: error: loop variable ‘it’ creates a copy from type ‘const std::pair<const unsigned int, std::unordered_set<std::__cxx11::basic_string<char> > >’ [-Werror=range-loop-construct]
  452 |                 for (const auto it : *eventIdFields) {
      |                                 ^~
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:452:33: note: use reference type to prevent copying
  452 |                 for (const auto it : *eventIdFields) {
      |                                 ^~
      |                                 &
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:454:37: error: loop variable ‘it2’ creates a copy from type ‘const std::__cxx11::basic_string<char>’ [-Werror=range-loop-construct]
  454 |                     for (const auto it2 : it.second) {
      |                                     ^~~
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:454:37: note: use reference type to prevent copying
  454 |                     for (const auto it2 : it.second) {
      |                                     ^~~
      |                                     &
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:482:33: error: loop variable ‘it2’ creates a copy from type ‘const std::pair<const std::__cxx11::basic_string<char>, std::unordered_set<std::__cxx11::basic_string<char> > >’ [-Werror=range-loop-construct]
  482 |                 for (const auto it2 : *filters) {
      |                                 ^~~
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:482:33: note: use reference type to prevent copying
  482 |                 for (const auto it2 : *filters) {
      |                                 ^~~
      |                                 &
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:484:37: error: loop variable ‘it3’ creates a copy from type ‘const std::__cxx11::basic_string<char>’ [-Werror=range-loop-construct]
  484 |                     for (const auto it3 : it2.second) {
      |                                     ^~~
/home/m/code/SysmonForLinux/sysmonLogView/sysmonLogView.cpp:484:37: note: use reference type to prevent copying
  484 |                     for (const auto it3 : it2.second) {
      |                                     ^~~
      |                                     &
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/sysmonLogView.dir/build.make:118: CMakeFiles/sysmonLogView.dir/sysmonLogView/sysmonLogView.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:143: CMakeFiles/sysmonLogView.dir/all] Error 2
make: *** [Makefile:103: all] Error 2
```

--- a/sysmonLogView/sysmonLogView.cpp
+++ b/sysmonLogView/sysmonLogView.cpp
@@ -449,9 +449,9 @@ void processCmdline(int argc, char *argv[],
                 }
                 printf("\n");
                 printf("Event Id Fields:\n");
-                for (const auto it : *eventIdFields) {
+                for (const auto& it : *eventIdFields) {
                     printf("  %d (", it.first);
-                    for (const auto it2 : it.second) {
+                    for (const auto& it2 : it.second) {
                         printf("%s ", it2.c_str());
                     }
                     printf(")\n");
@@ -479,9 +479,9 @@ void processCmdline(int argc, char *argv[],
                     printf("-1\n");
                 }
                 printf("Filters:\n");
-                for (const auto it2 : *filters) {
+                for (const auto& it2 : *filters) {
                     printf("  %s:\n", it2.first.c_str());
-                    for (const auto it3 : it2.second) {
+                    for (const auto& it3 : it2.second) {
                         printf("    %s\n", it3.c_str());
                     }
                 }

--- a/test/linuxRules.cpp
+++ b/test/linuxRules.cpp
@@ -150,7 +150,7 @@ TEST( Rules, WideString )
         { "\xc2\x90\xf0\x92\x8d\x85 ", {0x0090, 0xd808, 0xdf45, ' ', 0}, 8, 5 }
     };
 
-    for( auto CONST test : WideStrings ) {
+    for( auto CONST& test : WideStrings ) {
         EXPECT_EQ( WideStrlen( test.str1 ), test.str1len );
         EXPECT_EQ( WideStrlen( test.str2 ), test.str2len );
         EXPECT_EQ( WideStrcmp( test.str1, test.str2 ) == 0, test.strEqual );
@@ -182,7 +182,7 @@ TEST( Rules, WideString )
         EXPECT_EQ( test.lower, WideTolower( test.upper ) );
     }
 
-    for( auto CONST conv : WideConv ) {
+    for( auto CONST& conv : WideConv ) {
         CHAR utf8[32];
         WCHAR utf16[32];
         EXPECT_EQ( UTF8toUTF16( NULL, conv.utf8str, 0 ), conv.utf16count );

